### DISCUSSION TITLE
Move the Elastic cluster from PIM server to Outsite?

### DISCUSSION THREAD

**Discussion by arulselvan008** [pimcore/pimcore#17419] [Q&A / Support]
Hi Team,@jiajia

Inquire about the possibility of moving our Elastic cluster from the PIM server to an external location. We are exploring options to optimize our infrastructure and believe that this migration could offer significant benefits.

Could you please provide information on the feasibility, potential impact, and any necessary steps involved in this process? Additionally, any insights on best practices or recommendations for a smooth transition would be greatly appreciated.

Thank you for your assistance and support.

Regards,
Arul

**Comment by markus-moser** [comment]
Hi Arul,

this shouldn't be a big problem. You need to ensure that the latency between the servers is low. As elasticsearch works HTTP based the differences between a local and remote installation is quite small. Just ensure that the server has access to the remote location.

For the migration of the data the easiest solution most of the time is to just reindex it with the provided commands by the various bundles. If you need to migrate data which cannot be reconstructed out of database data with some commands (e.g. for statistics explorer data) you could also use native reinding from a remote location:
https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html#reindex-from-remote

From my experience a local elasticsearch instance sometimes makes some troubles because of memory problems or similar. Therefore a managed service/cloud-based elasticsearch instance could be a good benefit.

Are you interested in any specific aspect in detail?

Best,
Markus
